tests/test_symbolics.py

```diff
@@ -587,6 +590,50 @@ def test_minmax_precision(dtype, expected):
     assert np.all(f.data == 6.0)
 
 
+@pytest.mark.parametrize('dtype,expected', [
+    (np.float32, "powf"),
+    (np.float64, "pow"),
+])
+def test_pow_precision(dtype, expected):
+    grid = Grid(shape=(5, 5), dtype=dtype)
+
+    f = Function(name="f", grid=grid)
+    g = Function(name="g", grid=grid)
+
+    eqn = Eq(f, g**1.5)
+
+    op = Operator(eqn)
+
+    g.data[:] = 4.0
+
+    op.apply()
+
+    assert expected in str(op)
+    assert np.all(f.data == 8.0)
+
+
+@pytest.mark.parametrize('dtype,expected', [
+    (np.float32, "absf"),
+    (np.float64, "abs"),
+])
+def test_abs_precision(dtype, expected):
+    grid = Grid(shape=(5, 5), dtype=dtype)
+
+    f = Function(name="f", grid=grid)
+    g = Function(name="g", grid=grid)
+
+    eqn = Eq(f, abs(g))
+
+    op = Operator(eqn)
+
+    g.data[:] = -1.0
+
+    op.apply()
+
+    assert expected in str(op)
```

Comment:
This seems like a bad way to test that the right thing is happening, especially since `pow` is a substring of `powf` and `abs` is a substring of `absf`. However, I have just cargo-cult'd the above test!

Reply:
that's OK